### PR TITLE
244 refactor application tile

### PR DESCRIPTION
**Pull request by shenghanAirWalk**
1. move application-tile component into `airview-compliance-ui` and refactor the code base to use material ui 5.
2. move the icon chip component from compliance table to a separate feature and add the storybook for it.
3. move the progress-bar component into `airview-compliance-ui` and add the storybook for it.
4. refactor the storybook to use material ui 5 for compliance table, control overview, progress bar, icon chip and application tile.

### CODE REVIEW

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-call-to-action-button.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const styles = useStyles();
+
+  return (
+    <Button
+      sx={styles.button}
+      className={classNames}
+      // component={Link}
+      href={href}
+      variant="contained"
+      size="small"
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-call-to-action-button.js:15]: The props API will need to allow the passing of some additional props to support routing:
- `component` prop - this should be assigned to the `component` prop of the `Button` component
- `linkProps`- this should be spread to the `Button` component

**Example:**

```jsx
export function ApplicationTileCallToActionButton({ href, label, classNames, component = "button", linkProps }) {
 return (
    <Button
      sx={styles.button}
      className={classNames}
      component={component}
      variant="contained"
      size="small"
      {...linkProps}
    >{label}</Button>
 )
}
```

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-call-to-action-button.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Button from "@mui/material/Button";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-call-to-action-button.js:3]: For consistency, import as a named import 

import { Button } from "@mui/material";

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-content-row.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Box from "@mui/material/Box";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-content-row.js:3]: For consistency, import as a named import: 

`import { Box } from "@mui/material";`

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-divider.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Divider from "@mui/material/Divider";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-divider.js:3]: For consistency, import as a named import: 

`import { Divider } from "@mui/material";`

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-header.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Box from "@mui/material/Box";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-header.js:3]: For consistency, import as a named import: 

`import { Box } from "@mui/material";`

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-title.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Typography from "@mui/material/Typography";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-title.js:3]: For consistency, import as a named import: 

`import { Typography } from "@mui/material";`

File: packages/airview-compliance-ui/src/features/application-tile/application-tile.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Box from "@mui/material/Box";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile.js:3]: For consistency, import as a named import: 

`import { Box } from "@mui/material";`

File: packages/airview-compliance-ui/src/features/progress-bar/progress-bar.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import PropTypes from "prop-types";
+import LinearProgress from "@mui/material/LinearProgress";
+import Typography from "@mui/material/Typography";
+import Box from "@mui/material/Box";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/progress-bar/progress-bar.js:5]: For consistency, import as a named import: 

`import { LinearProgress, Typography, Box } from "@mui/material";`

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-content.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    root: {
+      padding: 2,
+    },
+    collapseToggleBtn: {
+      display: "block",
+      margin: "0 auto",
+      padding: 0,
+    },
+    collapseWrapperInner: {
+      paddingTop: 0,
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-content.js:77]: This is missing logic to added padding, see [here](https://github.com/AirWalk-Digital/airview-frontend/blob/70abcb226609eee1e89c660351aa2fa964eccf88/src/components/application-tile/application-tile-content.js#L77)

File: packages/airview-compliance-ui/src/stories/application-tile/application-tile.stories.js [hunk 10]
```diff
@@ -0,0 +1,320 @@
+import React from "react";
+import { userEvent, within } from "@storybook/testing-library";
+import WarningIcon from "@mui/icons-material/Warning";
+import Typography from "@mui/material/Typography";
+import { Box } from "@mui/material";
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/stories/application-tile/application-tile.stories.js:5]: Fo consistency please import as named:

`import { Typography, Box } from "@mui/material";`

**Review by BinaryJim — CHANGES_REQUESTED** [review]
Good effort, a few observations that will need addressing and then I think we're good to merge.

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-content.js [hunk 11]
```diff
@@ -74,7 +72,7 @@ function useApplicationTileContentStyles() {
       padding: 0,
     },
     collapseWrapperInner: {
```
Comment (BinaryJim) [inline packages/airview-compliance-ui/src/features/application-tile/application-tile-content.js:74]: SX API refactor is preventing these styles being applied to the correct DOM node. Can you update to the example below, which will fix the layout error in the UI.


```jsx
collapseWrapperInner: {
   "& .MuiCollapse-wrapperInner": {
      paddingTop: collapsible ? 2 : 0,
    },
 }
```

**Review by BinaryJim — CHANGES_REQUESTED** [review]
See required change due to CSS styles not being correctly applied.

File: packages/airview-compliance-ui/src/features/application-tile/application-tile-content.js (hunk 11 above)
Comment (shenghanAirWalk) [reply packages/airview-compliance-ui/src/features/application-tile/application-tile-content.js:74]: thank you so much! it is actually working now

Comment (BinaryJim) [reply packages/airview-compliance-ui/src/features/application-tile/application-tile-content.js:74]: Welcome :)

**Review by BinaryJim — APPROVED** [review]
Nice work, good to merge!

**Pull request merged**